test/distributed/_composable/fsdp/test_fully_shard_state_dict.py

```diff
@@ -148,9 +148,14 @@ def _test_state_dict_save_load(self, model: nn.Module):
                 param.to_local(),
                 torch.ones_like(param.to_local()) * new_fill_value,
             )
-            self.assertEqual(
```

Comment:
The assertion was too strict before, and our implementation was actually incorrect since we did not re-pad the local tensor.